### ISSUE TITLE
Still can't debug an app more complicated than the blink example (IDFGH-9526)

### ISSUE THREAD

**Issue by JBHunterI** [espressif/esp-idf#10880]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v4.4.4

### Operating System used.

Windows

### How did you build your project?

Eclipse IDE

### If you are using Windows, please specify command line type.

CMD

### What is the expected behavior?

Start debugging my app based on FreeRTOS IoT Reference code in the ESP-IDF Eclipse Plugin and see it stop at the automatic breakpoint at the beginning of app_main().  If not, at the very least be able to hit the Pause tool to get control of it, add another breakpoint in app_main()'s infinite loop, then hit the Play tool to start it running again.

### What is the actual behavior?

Doesn't stop at app-main()'s automatic temporary breakpoint, and can stop it with the Pause tool, but adding a breakpoint in app_main()'s infinite loop and the hitting the Play tool shows this in the idf.py monitor when it starts-up again:

...
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).

Setting breakpoint at 0x4204825e and returning...
0x4204825e: esp_vApplicationIdleHook at C:/App/ESP/IDF/esp-idf-v4.4.4/components/esp_system/freertos_hooks.c:50 (discriminator 2)

Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).

Setting breakpoint at 0x4204825e and returning...
0x4204825e: esp_vApplicationIdleHook at C:/App/ESP/IDF/esp-idf-v4.4.4/components/esp_system/freertos_hooks.c:50 (discriminator 2)

Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).
...

### Steps to reproduce.

See the description in the previous 2 sections above.

### Build or installation Logs.

```plain
See the description in the previous 2 sections above.
```

### More Information.

See previously closed issue Unable to upgrade ESP-IDF v4.4.3 to latest OpenOCD v0.11.0-esp32-20221026 tool (IDFGH-9242) #10628 for history.

**Comment by JBHunterI** [comment]
In case anyone cares, I verified my ESP32-C3-DevKitM-1 had all the "green NC" labelled changes from its schematic applied (swapped R1 & R4 for R2 & R3, added C4, C5, & R14, removed R21 & R22). Blink example still seems to behave correctly when debugging it (initialy stops at beginning of app_main(), resumes with "Play" tool, stops with "Pause" tool, resumes again with "Play", stops at added breakpoints, even while it's running, doesn't stop when they're removed).

Updated my IDF directory to latest from release branch:
C:\App\ESP\IDF\esp-idf-v4.4.4> git describe --tags
v4.4.4-148-g4c2afac355

However, same Eclipse debugger still doesn't stop at app_main() when trying to attach to my app, and panics when trying to resume it:
...
I (15041) temp_sub_pub_and_led_control_demo: Task TempSubPubLED waiting for publish 2 to complete.
I (15151) coreMQTT: Ack packet deserialized with result: MQTTSuccess.
I (15151) coreMQTT: State record updated. New state=MQTTPublishDone.
I (15151) temp_sub_pub_and_led_control_demo: Rx'ed ack for QoS1 publish from Tx to /filter/TempSubPubLED (P3:F0).
I (15191) coreMQTT: De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
I (15191) coreMQTT: State record updated. New state=MQTTPubAckSend.
I (15191) temp_sub_pub_and_led_control_demo: Received incoming publish message {"temperatureSensor":{ "taskName": "TempSubPubLED", "temperatureValue": 28.255800, "iteration": 2}}
I (18771) temp_sub_pub_and_led_control_demo: Sending publish request to agent with message "{"temperatureSensor":{ "taskName": "TempSubPubLED", "temperatureValue": 28.255800, "iteration": 3}}" on topic "/filter/TempSubPubLED"
I (18771) coreMQTT: Publishing message to /filter/TempSubPubLED.

I (18771) temp_sub_pub_and_led_control_demo: Task TempSubPubLED waiting for publish 3 to complete.
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).

Setting breakpoint at 0x42049090 and returning...
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).

Setting breakpoint at 0x42049090 and returning...
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).
...

Results are identical if I use the unmodified [Featured FreeRTOS IoT Integration ](https://www.freertos.org/featured-freertos-iot-integration-targeting-an-espressif-esp32-c3-risc-v-mcu/#getting-started) demo my app is based on, with only the temp_sub_pub_and_led_control_demo enabled to run through the USB Serial/JTAG interface of my ESP32-C3-DevKitM-1.

**Comment by JBHunterI** [comment]
In case anyone cares, I verified my ESP32-C3-DevKitM-1 had all the "green NC" labelled changes from its schematic applied (swapped R1 & R4 for R2 & R3, added C4, C5, & R14, removed R21 & R22). Blink example still seems to behave correctly when debugging it (initialy stops at beginning of app_main(), resumes with "Play" tool, stops with "Pause" tool, resumes again with "Play", stops at added breakpoints, even while it's running, doesn't stop when they're removed).

Updated my IDF directory to latest from release branch:
C:\App\ESP\IDF\esp-idf-v4.4.4> git describe --tags
v4.4.4-148-g4c2afac355

However, same Eclipse debugger still doesn't stop at app_main() when trying to attach to my app, and panics when trying to resume it:
...
I (15041) temp_sub_pub_and_led_control_demo: Task TempSubPubLED waiting for publish 2 to complete.
I (15151) coreMQTT: Ack packet deserialized with result: MQTTSuccess.
I (15151) coreMQTT: State record updated. New state=MQTTPublishDone.
I (15151) temp_sub_pub_and_led_control_demo: Rx'ed ack for QoS1 publish from Tx to /filter/TempSubPubLED (P3:F0).
I (15191) coreMQTT: De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
I (15191) coreMQTT: State record updated. New state=MQTTPubAckSend.
I (15191) temp_sub_pub_and_led_control_demo: Received incoming publish message {"temperatureSensor":{ "taskName": "TempSubPubLED", "temperatureValue": 28.255800, "iteration": 2}}
I (18771) temp_sub_pub_and_led_control_demo: Sending publish request to agent with message "{"temperatureSensor":{ "taskName": "TempSubPubLED", "temperatureValue": 28.255800, "iteration": 3}}" on topic "/filter/TempSubPubLED"
I (18771) coreMQTT: Publishing message to /filter/TempSubPubLED.

I (18771) temp_sub_pub_and_led_control_demo: Task TempSubPubLED waiting for publish 3 to complete.
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).

Setting breakpoint at 0x42049090 and returning...
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).

Setting breakpoint at 0x42049090 and returning...
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0).
...

Results are identical if I use the unmodified [Featured FreeRTOS IoT Integration ](https://www.freertos.org/featured-freertos-iot-integration-targeting-an-espressif-esp32-c3-risc-v-mcu/#getting-started) demo my app is based on, with only the temp_sub_pub_and_led_control_demo enabled to run through the USB Serial/JTAG interface of my ESP32-C3-DevKitM-1.
